OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -498,22 +504,22 @@ bool HasAdequateAirUnitReloadBuildings(ActorInfo actorInfo)
 				case BuildingType.Refinery:
 
 					// Try and place the refinery near a resource field
-					var nearbyResources = Map.FindTilesInCircle(baseCenter, Info.MaxBaseRadius)
+					var nearbyResources = Map.FindTilesInAnnulus(baseCenter, Info.MinBaseRadius, Info.MaxBaseRadius)
 						.Where(a => resourceTypeIndices.Get(Map.GetTerrainIndex(a)))
-						.Shuffle(Random);
+						.Shuffle(Random).ToList().Take(Info.MaxResourceCellsToCheck);
```

Comment:
You can drop the `ToList` now, so we're only going to enumerate it once. Sorry for the back and forth.